File: wk02-linked-list-complete-search/#11502 세 개의 소수 문제/김서로.py
import sys
import math
input = sys.stdin.readline

def is_prime(num):
    for i in range(2, int(math.sqrt(num)) + 1):
        if num % i == 0:
            return False
    return True

def three_odd_num(num):
    answer = []
    for i in range(2, num):
        for j in range(2, num):
            if num-i-j > 0 and prime[i] and prime[j] and prime[num-i-j]:
                answer.append(i)
                answer.append(j)
                answer.append(num-i-j)
                answer = list(map(str, sorted(answer)))
                return " ".join(answer)
    return 0

# 소수 미리 구해두기
prime = [False] * 1000

for i in range(2, 1000):
    prime[i] = is_prime(i)

T = int(input())

for _ in range(T):
    k = int(input())
    print(three_odd_num(k))
        

File: wk02-linked-list-complete-search/#11502 세 개의 소수 문제/김서로.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main {

	private static BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
    // 소수인지 여부를 boolean으로 저장
	private static boolean[] prime = new boolean[1000];
    
    // 소수인지 판단
	private static boolean isPrime(int num) {
		for (int i = 2; i < (int) Math.sqrt(num) + 1; i++) {
			if (num % i == 0) {
				return false;
			}
		}
		return true;
	}
    
    // 홀수를 세 개의 소수로 출력
	private static void printThreePrime(int num) {
		for (int i = 2; i < num; i++) {
			for (int j = 2; j < num; j++) {
				if (num - i - j > 0 && prime[i] && prime[j] && prime[num - i - j]) {
					StringBuilder output = new StringBuilder();
					output.append(i).append(" ").append(j).append(" ").append(num - i - j);
					System.out.println(output);
					return;
				}
			}
		}
		System.out.println(0);
	}

	public static void main(String[] args) throws Exception {
		// 소수 미리 구해두기
		for (int i=2; i < 1000; i++) {
			prime[i] = isPrime(i);
		}
		
		int T = Integer.parseInt(input.readLine());
		
		for (int i=0; i < T; i++) {
			int k = Integer.parseInt(input.readLine());
			printThreePrime(k);
		}
	}

}
